calculator_test.go

```diff
@@ -2,23 +2,558 @@ package calculator_test
 
 import (
 	"calculator"
+	"fmt"
+	"math"
+	"math/rand"
 	"testing"
 )
 
+func closeEnough(
+	roundResult, divisionResult, tolerance float64,
+) bool {
+	fmt.Println(
+		"aaa",
+		math.Abs(roundResult-divisionResult) <= tolerance,
+	)
+	return math.Abs(roundResult-divisionResult) <= tolerance
+}
+
+func TestCloseEnough(t *testing.T) {
+	testCases := []struct {
+		a, b, roundedResult, tolerance float64
+		want                           bool
+	}{
+		{a: 1.66666666, b: 1.6666, tolerance: 0.0001, want: true},
+		{a: 0.33333333, b: 0.333, tolerance: 0.001, want: true},
+		{a: 0.66666666, b: 0.66, tolerance: 0.001, want: false},
+	}
+
+	for _, tc := range testCases {
+		got := closeEnough(tc.a, tc.b, tc.tolerance)
+
+		if tc.want != got {
+			t.Errorf(
+				"Result(%g) is not close enough(%.5f)",
+				tc.b,
+				tc.tolerance,
+			)
+		}
+	}
+}
+
+func TestAddSubtractMultiply(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		operator   func(float64, float64, ...float64) float64
+		a, b, want float64
```

Comment:
These are `float64`, but you don't actually have any fractional cases! That seems like a good way to make the test cases a bit more demanding, doesn't it?

Reply:
Added in f292791 and fixed tests in 5ea46f7